deployment/setup_20.04.sh

```diff
@@ -1,164 +1,282 @@
-then
-exit 1
-fi
-fi
+set -eu -o pipefail
+trap exit_handler EXIT
 
-read -p 'Would you like to install postgres and redis?(Answer no if you plan to use external services): ' install_pg_redis
+function exit_handler() {
+  if [ "$?" -ne 0 ]; then
+   echo -en "\nSome error has occured. Check '/var/log/chatwoot-setup.log' for details.\n"
+   exit 1
+  fi
+}
 
-if [ $install_pg_redis == "no" ]
-then
-echo "***** Skipping pg and redis installation. ****"
+if [ "$(id -u)" -ne 0 ]; then
+  echo 'This script should be run as root.' >&2
+  exit 1
 fi
 
-apt update && apt upgrade -y
-apt install -y curl
-curl -sL https://deb.nodesource.com/setup_12.x | bash -
-curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
-echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-apt update
-
-apt install -y \
-    git software-properties-common imagemagick libpq-dev \
-    libxml2-dev libxslt1-dev file g++ gcc autoconf build-essential \
-    libssl-dev libyaml-dev libreadline-dev gnupg2 \
-    postgresql-client redis-tools \
-    nodejs yarn patch ruby-dev zlib1g-dev liblzma-dev \
-    libgmp-dev libncurses5-dev libffi-dev libgdbm6 libgdbm-dev sudo
-
-if [ $install_pg_redis != "no" ]
-then
-apt install -y postgresql postgresql-contrib redis-server
+if [[ -z "$1" ]]; then
+  BRANCH="master"
+else
+  BRANCH="$1"
 fi
 
-if [ $configure_webserver == "yes" ]
-then
-apt install -y nginx nginx-full certbot python3-certbot-nginx
-fi
+function get_domain_info() {
+  read -rp 'Enter your sub-domain to be used for Chatwoot (chatwoot.domain.com for example) : ' domain_name
+  read -rp 'Enter the email LetsEncrypt can use to send reminders when your SSL certificate is up for renewal: ' le_email
+  cat << EOF
+
+This script will try to generate SSL certificates via LetsEncrypt and serve chatwoot at
+https://$domain_name. Proceed further once you have pointed your DNS to the IP of the instance.
 
-adduser --disabled-login --gecos "" chatwoot
-
-gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
-gpg2 --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
-curl -sSL https://get.rvm.io | bash -s stable
-adduser chatwoot rvm
-
-if [ $install_pg_redis != "no" ]
-then
-pg_pass=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 15 ; echo '')
-sudo -i -u postgres psql << EOF
-\set pass `echo $pg_pass`
-CREATE USER chatwoot CREATEDB;
-ALTER USER chatwoot PASSWORD :'pass';
-ALTER ROLE chatwoot SUPERUSER;
-UPDATE pg_database SET datistemplate = FALSE WHERE datname = 'template1';
-DROP DATABASE template1;
-CREATE DATABASE template1 WITH TEMPLATE = template0 ENCODING = 'UNICODE';
-UPDATE pg_database SET datistemplate = TRUE WHERE datname = 'template1';
-\c template1
-VACUUM FREEZE;
 EOF
+  read -rp 'Do you wish to proceed? (yes or no): ' exit_true
+  if [ "$exit_true" == "no" ]
```

Comment:
Y/N is better right?

Reply:
I think It's better to be verbose here so that the user stops for a moment and makes a conscious choice.